### PR TITLE
Implementfilesend

### PR DESCRIPTION
**Pull request by kfless12**
Implemented EJS and sequlize together with the server.
Made several adjustments to the sequelize file snippets and naming structures of some of the snippets "filename" (for working purposes)
connected the download button to send over the boiler plate file as a download

### CODE REVIEW

File: server/api/zippedBoilerPlate.js [hunk 1]
```diff
@@ -1,72 +1,85 @@
 const zippedBoilerPlate = require('express').Router();
+const { models: { Code }} = require('../db')
 const ejs = require('ejs')
 const fs = require('fs')
 const zip = new require('node-zip')()
 const path = require('path')
 
 //helper function designed to deal with multiple files in a directory folder
 function zipHelper(fileString, arrObject){
```
Comment (king-ellie) [inline server/api/zippedBoilerPlate.js]: zipHelper should be async, and all zipHelper calls should have await in front of them

File: server/api/zippedBoilerPlate.js [hunk 2]
```diff
@@ -1,72 +1,85 @@
...
 }
 
-zippedBoilerPlate.get('/', async (req, res, next) => {
+zippedBoilerPlate.post('/', async (req, res, next) => {
   try {
-
-    let appjs = createFile('App', {router: true, redux: true}, 'js')
+    let boiler = req.body
+    //need to fix the issues here with wrong datatypes being passed through for whatever reason
+    let appjs = createFile('R2', boiler)
```
Comment (king-ellie) [inline server/api/zippedBoilerPlate.js]: I believe appjs should only be created if boiler.react is truthy. In other words, if a user says react: false, all they should receive for the front end is a client folder with index.html in it

File: server/api/zippedBoilerPlate.js [hunk 3]
```diff
@@ -1,72 +1,85 @@
...
-    arrObject.forEach(file => zip.file(`${fileString}${file.name}`, file.contents))
-
+    Promise.all(arrObject).then((results)=>{
+      results.forEach(file => zip.file(`${fileString}${file.name}`, file.contents))
+    })
 }
 
-function createFile(filename, variable, ext){
-    let name = `${filename}.${ext}`
+async function createFile(key, variable){
```
Comment (king-ellie) [inline server/api/zippedBoilerPlate.js:13]: All calls of createFile need to have "await" in front of them

**Review by king-ellie — CHANGES_REQUESTED** [review]
Awesome work, thank you so much! Mostly the zipHelper and createFile calls just need to be updated to async/have "await" in front of them. Seriously amazing how you walked through the logic to build these zip files

**Review by king-ellie — APPROVED** [review]

**Review by mpiech1813 — APPROVED** [review]

**Pull request merged**